src/ryu_faucet/org/onfsdn/faucet/watcher.py

```diff
@@ -301,6 +335,38 @@ class GaugeFlowTablePoller(GaugePoller):
 
     def __init__(self, conf, logname):
         super(GaugeFlowTablePoller, self).__init__(conf, logname)
+        # Database specific config file read
+        self.db_config = os.getenv(
+            'GAUGE_DB_CONFIG', '/etc/ryu/faucet/gauge_db.yaml')
+        self.db_enabled = False
+        self.flow_database = None
+        self.switch_database = None
+        self.conn = None
+        with open(self.db_config, 'r') as stream:
+            data = yaml.load(stream)
+            if data['database']:
+                self.db_enabled = True
+                self.conn_string = "driver={0};server={1};port={2};" \
+                                   "uid={3};pwd={4}".format(
+                    data['driver'], data['db_ip'], str(data['db_port']),
+                    str(data['db_username']), str(data['db_password'])
+                )
+                nsodbc = nsodbc_factory()
+                self.conn = nsodbc.connect(self.conn_string)
+                self.switch_database, exists = self.conn.create(
+                    data['switches_doc'])
+                # Create database specific views for querying
+                if not exists:
+                    init_switch_db(self.switch_database)
+
+                self.flow_database, exists = self.conn.create(
+                    data['flows_doc'])
+                # Create database specific views for querying
+                if not exists:
+                    init_flow_db(self.flow_database)
+                self.db_conf_data = data
+        # TODO: DB update counter can be made configurable
+        self.db_update_counter = 5
```

Comment:
This should be in a GaugeFlowTableCouchPoller class, rather than this one. Also I think when this is initialised is a more appropriate time to initialise the database.